packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts

```diff
@@ -464,6 +472,60 @@ export class StripeWebhookHandler extends StripeHandler {
     await this.sendSubscriptionPaymentFailedEmail(invoice);
   }
 
+  /**
+   * Handle `invoice.upcoming` Stripe wehbook events.
+   */
+  async handleInvoiceUpcomingEvent(request: AuthRequest, event: Stripe.Event) {
+    const invoice = event.data.object as Stripe.Invoice;
+    if (invoice.billing_reason !== 'upcoming') {
+      // Send payment failure emails only when processing a subscription renewal.
+      return;
+    }
+    const customer = await this.stripeHelper.expandResource(
+      invoice.customer,
+      CUSTOMER_RESOURCE
+    );
+
+    if (
+      customer?.deleted ||
+      !customer?.invoice_settings?.default_payment_method
+    ) {
+      return;
+    }
+
+    const { card } = await this.stripeHelper.expandResource(
+      customer?.invoice_settings.default_payment_method,
+      PAYMENT_METHOD_RESOURCE
+    );
+
+    const currentDate = new Date();
+
+    if (
+      !(
+        card?.exp_month === currentDate.getMonth() + 1 &&
```

Comment:
Maybe use a type parameter on the `expandResource` call so that `card` isn't `any`

Is the `+ 1` because the value returned by `getMonth` is zero indexed, or is this checking for expiry in next month?  I cannot tell simply from reading this.

Reply:
Yup that's right. Looks like getMonth is zero indexed.

I created 2 new vars `currentMonth` and `currentYear` that should make it more clear. What do you think?

Reply:
So I changed this logic around a little so that we look for any expired cards, not just the ones that expire this current month.